ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -294,7 +288,45 @@ let get_initial_guest_metrics (lookup : string -> string option)
     | None ->
         []
   in
-  let pv_drivers_version = to_map pv_drivers_version
+  (* enumerate all driver versions from xenstore, which are stored like
+     drivers/0 = "XenServer XENBUS 9.1.9.105 "
+     drivers/1 = "XenServer XENVBD 9.1.8.79 "
+     drivers/2 = "XenServer XENVIF 9.1.12.101 "
+     drivers/3 = "XenServer XENIFACE 9.1.10.87 "
+     drivers/4 = "XenServer XENNET 9.1.7.65 "
+
+     (see the format specified in xenstore-paths)
+  *)
+  let get_windows_driver_versions () =
+    (* Only look into directories that are numbers (indices) *)
+    let filter_dirs subdirs =
+      List.filter_map
+        (fun x ->
+          match int_of_string_opt x with
```

Comment:
In theory, this would match `0x011`, `-3`, and `0o123` as well. To avoid that you would need `"%u%!"` in scanf.